### DISCUSSION TITLE
Building liboqs with OpenSSL 3. vs OpenSSL 1.1.1

### DISCUSSION THREAD

**Discussion by fj-blanco** [open-quantum-safe/openssl#444] [General]
Up until recently, I had successfully managed to build liboqs using my default OpenSSL 3.0.2 installation on Ubuntu 22.04.2 LTS. Following the build, I would then install liboqs into the Open Quantum Safe's OpenSSL 1.1.1. This setup worked smoothly for me.

However, following the liboqs commit [871f9e26d1763f512255f7e734f7a126638063c6](https://github.com/open-quantum-safe/liboqs/commit/871f9e26d1763f512255f7e734f7a126638063c6), this method appears to be no longer valid because undefined references to functions such as EVP_MD_fetch, EVP_CIPHER_fetch, EVP_MD_free, and EVP_CIPHER_free in the file ossl_helpers.c.

Although I'm uncertain whether building liboqs with OpenSSL 3.* is the best approach (as I didn't encounter any issues in my benchmarks), it appears to no longer be an option, and OpenSSL 1.1.1 seems to be required.

Currently, I'm transitioning to using the Provider and OpenSSL 3. I'm sharing this for reference and to potentially benefit others who might encounter a similar scenario.

**Comment by baentsch** [comment]
Thanks for sharing this. This is exactly #442 and addressed by #443 (now merged).
